### PR TITLE
[.NET] ActivityWatcher.GetServerLocation() Http optimization

### PR DESCRIPTION
**Pull request by Redusofficial**
Changed GetServerLocation() to only make a single request to ipinfo.io to retrieve the needed location information

### CODE REVIEW

**Comment by pizzaboxer** [comment]
@bluepilledgreat Did we already know about this? I could've sworn there was a reason we avoided using this when we first implemented it.

**Comment by bluepilledgreat** [comment]
> @bluepilledgreat Did we already know about this? I could've sworn there was a reason we avoided using this when we first implemented it.

¯\_(ツ)_/¯

File: Bloxstrap/Integrations/ActivityWatcher.cs [hunk 1]
```diff
@@ -284,9 +284,11 @@ public async Task<string> GetServerLocation()
 
             try
             {
-                locationCity = await App.HttpClient.GetStringAsync($"https://ipinfo.io/{ActivityMachineAddress}/city");
-                locationRegion = await App.HttpClient.GetStringAsync($"https://ipinfo.io/{ActivityMachineAddress}/region");
-                locationCountry = await App.HttpClient.GetStringAsync($"https://ipinfo.io/{ActivityMachineAddress}/country");
+                locationInformation = await Http.GetJson<IPInfoResponse>($"https://ipinfo.io/{ActivityMachineAddress}/json");
```
Comment (bluepilledgreat) [inline Bloxstrap/Integrations/ActivityWatcher.cs:287]: `locationInformation` is not properly initialised and will cause an error on build.

File: Bloxstrap/Models/IPInfoResponse.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+namespace Bloxstrap.Models
+{
+    public class IPInfoResponse
+    {
+        [JsonPropertyName("city")]
+        public string City { get; set; }
```
Comment (bluepilledgreat) [inline Bloxstrap/Models/IPInfoResponse.cs:6]: Make all the string properties nullable to avoid build warnings.

**Review by bluepilledgreat — CHANGES_REQUESTED** [review]
Please properly test your code to see if it works and even builds

File: Bloxstrap/Models/IPInfoResponse.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+        [JsonPropertyName("country")]
+        public string Country { get; set; }
+
+        [JsonPropertyName("region")]
+        public string Region { get; set; }
+
+        [JsonPropertyName("ip")]
+        public string IP { get; set; }
```
Comment (bluepilledgreat) [inline Bloxstrap/Models/IPInfoResponse.cs:15]: Only include properties in the response class that are used by the original code.

**Review by bluepilledgreat — CHANGES_REQUESTED** [review]

**Comment by bluepilledgreat** [comment]
Please actually bother to build and test the build or I'm closing the PR.

**Comment by Redusofficial** [comment]
I'm currently doing that

**Comment by pizzaboxer** [comment]
Looks good, let me make some more changes.

**Comment by bluepilledgreat** [comment]
> Looks good, let me make some more changes.

Have you tested it?

**Comment by pizzaboxer** [comment]
Yeah, it all works.

**Review by pizzaboxer — APPROVED** [review]

**Review by bluepilledgreat — APPROVED** [review]

**Pull request merged**